cmd/libs/go2idl/conversion-gen/generators/conversion.go

```diff
@@ -693,13 +705,22 @@ func (g *genConversion) doStruct(inType, outType *types.Type, sw *generator.Snip
 				sw.Do("out.$.name$ = $.outType|raw$(in.$.name$)\n", args)
 			}
 		case types.Map, types.Slice, types.Pointer:
+			if g.isDirectlyAssignable(m.Type, outMember.Type) {
+				sw.Do("out.$.name$ = in.$.name$\n", args)
```

Comment:
We don't need to cast on assignments where two objects share the same underlying type, i.e. aliases (which is what isDirectlyAssignable checks).